src/services/lnd/onchain-service.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  UnknownOnChainServiceError,
+  OnChainServiceUnavailableError,
+} from "@domain/bitcoin/onchain"
+import { toSats } from "@domain/bitcoin"
+import { getHeight, getChainTransactions } from "lightning"
+import { getActiveLnd } from "./utils"
+
+export const MakeOnChainService = (
+  decoder: TxDecoder,
+): IOnChainService | OnChainServiceError => {
+  let lnd: AuthenticatedLnd
+
+  try {
+    ;({ lnd } = getActiveLnd())
```

Comment:
you must use pubkey to get lnd because addresses are generated in the specific lnd node. @nicolasburtey could you please confirm/validate?

Reply:
For this use case this is the code that existed in the old code base the multi-lnd use case is not yet fully supported

Reply:
no, it uses the pubkey https://github.com/GaloyMoney/galoy/blob/main/src/core/on-chain/index.ts#L423

Reply:
I know we dont support dynamic or disposal lnds but we are deploying 2 lnds (or at least that is what I see in charts). Please ask nicolas and if you want to keep this code check/validate in the db that there are only one pubkey in onchain addresses

Reply:
Spoke to @nicolasburtey and confirmed that this is currently _not_ a blocker for this PR